racket/src/racket/src/jitinline.c

```diff
@@ -1283,6 +1284,12 @@ int scheme_generate_inlined_unary(mz_jit_state *jitter, Scheme_App2_Rec *app, in
   } else if (IS_NAMED_PRIM(rator, "path?")) {
     generate_inlined_type_test(jitter, app, SCHEME_PLATFORM_PATH_KIND, SCHEME_PLATFORM_PATH_KIND, 0, for_branch, branch_short, dest);
     return 1;
+  } else if (IS_NAMED_PRIM(rator, "hash?")) {
+    generate_inlined_type_test(jitter, app, scheme_hash_table_type, scheme_hash_tree_indirection_type, 1, for_branch, branch_short, dest);
```

Comment:
Should it also cover `scheme_bucket_type`?

Also, those are not in a sequence with `scheme_hash_table_type` et al. Should I move them in `stypes.h`, or write a more complex JIT function?